src/edit.c

```diff
@@ -2926,34 +2932,58 @@ stuff_inserted(
     // may want to stuff the command character, to start Insert mode
     if (c != NUL)
 	stuffcharReadbuff(c);
-    if ((esc_ptr = vim_strrchr(ptr, ESC)) != NULL)
-	*esc_ptr = NUL;	    // remove the ESC
+
+    // look for the last ESC in 'insert'. if we find one, replace it with NUL;
+    // remember where it is so we can restore it later.
+    for (esc_ptr = insert.string + insert.length;
```

Comment:
Could it be like this?
```suggestion
    for (esc_ptr = insert.string + insert.length - 1;
```

Reply:
It could.
If `insert.length` is 0, `esc_ptr` would point to just before the start of `insert.string`. Is that going to work for all platforms? Not sure that's a good idea.

Reply:
No, correct it together with zeertzjq's comment immediately below.

```c
    for (esc_ptr = insert.string + insert.length - 1;
		esc_ptr >= insert.string;
		MB_PTR_BACK(insert.string, esc_ptr))
```

You should feel it's bad practice to access array data with `insert.string + insert.length` in C.